geomstats/geometry/discrete_curves.py

```diff
@@ -1907,28 +2231,48 @@ def dist(self, point_a, point_b, n_times=20, threshold=1e-3):
             Discrete curve.
         point_b : array-like, shape=[k_sampling_points, ambient_dim]
             Discrete curve.
+        method : str, {"iterative horizontal projection", "dynamic programming"}
+            Type of method to use.
+            Optional, default: "iterative horizontal projection".
         n_times: int
             Number of times used to discretize the horizontal geodesic.
             Optional, default: 20.
         threshold: float
-            Stop criterion used in the algorithm to compute the horizontal
-            geodesic.
+            Stop criterion used in the algorithm to compute the horizontal geodesic.
             Optional, default: 1e-3.
+        n_discretization : int
+            Number of subintervals in which the reparametrization is linear.
+            Optinal, default: 100.
+        max_slope : int
+            Maximum slope allowed for a reparametrization.
+            Optional, default: 6.
 
         Returns
         -------
         quotient_dist : float
             Quotient distance between the two curves represented by point_a and point_b.
         """
-        horizontal_path = self.geodesic(
-            initial_point=point_a, end_point=point_b, threshold=threshold
-        )
-        times = gs.linspace(0.0, 1.0, n_times)
-        horizontal_geod = horizontal_path(times)
-        horizontal_geod_velocity = n_times * (
-            horizontal_geod[:-1] - horizontal_geod[1:]
-        )
-        velocity_norms = self.fiber_bundle.total_space.metric.norm(
-            horizontal_geod_velocity, horizontal_geod[:-1]
-        )
-        return gs.sum(velocity_norms) / n_times
+        if method == "iterative horizontal projection":
+
+            horizontal_path = self.geodesic(
+                initial_point=point_a, end_point=point_b, threshold=threshold
+            )
+            times = gs.linspace(0.0, 1.0, n_times)
+            horizontal_geod = horizontal_path(times)
+            horizontal_geod_velocity = n_times * (
+                horizontal_geod[:-1] - horizontal_geod[1:]
+            )
+            velocity_norms = self.fiber_bundle.total_space.metric.norm(
+                horizontal_geod_velocity, horizontal_geod[:-1]
+            )
+            return gs.sum(velocity_norms) / n_times
+
+        if method == "dynamic programming" :
+            results = self.fiber_bundle._dynamic_programming(
+                point_a, point_b, n_discretization, max_slope)
+            return results["distances"]
+
+        else :
+            raise AssertionError(
+                "Method not implemented"
```

Comment:
Maybe be more specific here: "The only methods implemented are iterative horizontal projection and dynamic programming."